lib/pdksync.rb

```diff
@@ -44,10 +47,147 @@ module PdkSync
   @default_pdksync_label = Constants::PDKSYNC_LABEL
   @git_platform = Constants::GIT_PLATFORM
   @git_base_uri = Constants::GIT_BASE_URI
+  @jenkins_platform = Constants::JENKINS_PLATFORM
   @git_platform_access_settings = {
     access_token: Constants::ACCESS_TOKEN,
     gitlab_api_endpoint: Constants::GITLAB_API_ENDPOINT
   }
+  @jenkins_platform_access_settings = {
+    jenkins_username: Constants::JENKINS_USERNAME,
+    jenkins_password: Constants::JENKINS_PASSWORD,
+    jenkins_api_endpoint: Constants::JENKINS_API_ENDPOINT
+  }
+
+  # convert duration from milliseconds to format h m s ms
+  def self.get_duration_hrs_and_mins(milliseconds)
+    return '' unless milliseconds
+    hours, milliseconds   = milliseconds.divmod(1000 * 60 * 60)
+    minutes, milliseconds = milliseconds.divmod(1000 * 60)
+    seconds, milliseconds = milliseconds.divmod(1000)
+    "#{hours}h #{minutes}m #{seconds}s #{milliseconds}ms"
+  end
+
+  # jenkins report
+  def self.jenkins_report_analisation(github_repo, build_id)
+    job_name = "forge-module_#{github_repo}_init-manual-parameters_adhoc"
```

Comment:
This is specific to puppet and will fail outside of puppet.

Reply:
Hey @logicminds ,
In the initial phase this task will be used internally. We'll add more functionalities in the next phase. Thanks!